be/src/env/env_s3.cpp

```diff
@@ -194,7 +195,15 @@ S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const ClientConfigurati
 }
 
 static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri) {
-    Aws::Client::ClientConfiguration config;
+    // Since aws sdk has changed the Aws::Client::ClientConfiguration default constructor to
+    // search for the region (where as before 1.8 it has been hard coded default of "us-east-1").
+    // Part of that change is looking through the ec2 metadata, which can take a long time.
+    // By having AWS_EC2_METADATA_DISABLED it avoids doing this search.
+    // For more details, please refer https://github.com/aws/aws-sdk-cpp/issues/1440
+    setenv("AWS_EC2_METADATA_DISABLED", "true", 1);
```

Comment:
Is `setenv` thread-safe?

Reply:
Can we skip searching region by constructing a `ClientConfiguration` instance by copying from a global variable?
```
static Aws::Client::ClientConfiguration s_config;
Aws::Client::ClientConfiguration config = s_config;
...
```

Reply:
> Can we skip searching region by creating a copy of a static variable of type Aws::Client::ClientConfiguration?
> 
> ```
> static Aws::Client::ClientConfiguration s_config;
> Aws::Client::ClientConfiguration config = s_config;
> ...
> ```

The config.endpointOverride field will be override if we use the same ClientConfiguration object for all endpoint.
I am thinking about cache static ClientConfiguration used endpoint as key.

Reply:
> > Can we skip searching region by creating a copy of a static variable of type Aws::Client::ClientConfiguration?
> > ```
> > static Aws::Client::ClientConfiguration s_config;
> > Aws::Client::ClientConfiguration config = s_config;
> > ...
> > ```
> 
> The config.endpointOverride field will be override if we use the same ClientConfiguration object for all endpoint. I am thinking about cache static ClientConfiguration use endpoint as key.

We are not using the same object, each time you call `Aws::Client::ClientConfiguration config = s_config;` a new object is created and the `endpointOverride` will not be shared among each other.

Reply:
> > > Can we skip searching region by creating a copy of a static variable of type Aws::Client::ClientConfiguration?
> > > ```
> > > static Aws::Client::ClientConfiguration s_config;
> > > Aws::Client::ClientConfiguration config = s_config;
> > > ...
> > > ```
> > 
> > 
> > The config.endpointOverride field will be override if we use the same ClientConfiguration object for all endpoint. I am thinking about cache static ClientConfiguration use endpoint as key.
> 
> We are not using the same object, each time you call `Aws::Client::ClientConfiguration config = s_config;` a new object is created and the `endpointOverride` will not be shared among each other.

LGTM，i will change the implementation